sdk/iot/provisioning/samples/src/paho_iot_provisioning_example.c

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <paho-mqtt/MQTTClient.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <az_iot_provisioning_client.h>
+#include <az_result.h>
+#include <az_span.h>
+
+// Service information
+#define GLOBAL_PROVISIONING_ENDPOINT_DEFAULT "ssl://global.azure-devices-provisioning.net:8883"
+#define GLOBAL_PROVISIONING_ENDPOINT_ENV "AZ_IOT_GLOBAL_PROVISIONING_ENDPOINT"
+#define ID_SCOPE_ENV "AZ_IOT_ID_SCOPE"
+
+// Device information
+#define REGISTRATION_ID_ENV "AZ_IOT_REGISTRATION_ID"
+// AZ_IOT_DEVICE_X509_CERT_PEM_FILE is the path to a PEM file containing the device certificate and
+// key as well as any intermediate certificates chaining to an uploaded group certificate.
+
+#define DEVICE_X509_CERT_PEM_FILE "AZ_IOT_DEVICE_X509_CERT_PEM_FILE"
+
+static char global_provisioning_endpoint[256] = { 0 };
```

Comment:
nit nit: here the `{ 0 }` aren't needed

Reply:
I like to have it clear that it's all zeros.

Reply:
no prob sounds good 👍